themes/default/content/resources/walkthrough-of-building-container-based-aws-lambda-functions/index.md

```diff
@@ -0,0 +1,75 @@
+
+        Today's example is in TypeScript, but Pulumi makes it easy to stand up infrastructure in your favorite languages including JavaScript, Python, C#, and Go - saving time over legacy tools like CloudFormation and Hashicorp Terraform.
+
+    # The video presenters
+    presenters:
+        - name: Lee Zen
+          role: VP of Engineering, Pulumi
+
+# This section contains the transcript for a video. It is optional.
+transcript: |
+    Hello, and welcome to another episode of Modern Infrastructure Wednesday. Today, we're going to be going over the demo that it built for showing how to use container-based Lambda functions. And so we're going to walk through that code in depth. Since in the video, it was like kind of a quick take on just how to build an application and kind of show the demo of the application, but it never really walked through it in depth. And so we're going to do that in this video today. If we kind of start at the beginning, we import a couple of libraries. We're building a TypeScript application today Pulumi obviously does support Go, .NET as well as Python in addition to Node.js. So, but here we're using TypeScript. So I create this bucket you know, simply just call, you know, a new bucket. Here, we just name it bucket, but obviously that's just the, the name we're using within Pulumi. With Pulumi's auto naming capabilities these actually ended up being transformed into suffix names and the actual cloud provider in this case AWS. This next line here, we're using this convenience library, it'll be AWSX, this is a Crosswalk for AWS.
+
+    It's a convenience library from Pulumi. And then here we just build and push image and it builds a the image and pushes it into a repository within ECR. And so actually, if we go to our console, the Pulumi console, we can actually see those resources here in our stack. And if I open this up we can actually see the name of the bucket is, you know, is suffix like this. And then same thing with I, if I go look at the sample app repository, we can actually see this particular repository in ECR.
+
+    Sorry, I updated something without publishing it, but here we can see that the repository names as well as the images, that's what I meant to do as long as the images that have been pushed into that repository. So that's going to how that works. And so again, this is all auto named as part of the bloomy. And so yeah, so we, we have the bucket, we have the image in order to execute a Lambda function, obviously that Lambda function needs to have an IAM role and permissions. And so we create that role here, thumbnail role, and we attach a policy called the AWS Lambda, full access policies, actually here, we using a managed policy arm that AWS provides. You'll note that this is actually an enum, you know and so, so that's how this is how we actually bring this in here.
```

Comment:
😂 "the bloomy"

```suggestion
    Sorry, I updated something without publishing it, but here we can see that the repository names as well as the images, that's what I meant to do as long as the images that have been pushed into that repository. So that's kind of how that works. And so again, this is all auto named as part of Pulumi. And so yeah, so we, we have the bucket, we have the image in order to execute a Lambda function, obviously that Lambda function needs to have an IAM role and permissions. And so we create that role here, thumbnail role, and we attach a policy called the AWS Lambda full-access policy. So actually here, we using a managed policy ARN that AWS provides. You'll note that this is actually an enum, you know and so, so that's how this is how we actually bring this in here.
```